Add Docker to Observability Pipelines setup guides (#18961)
DataDog/documentation: content/en/observability_pipelines/setup/splunk.md

### Provider-specific requirements
{{< tabs >}}
{{% tab "Docker" %}}

> Ensure that your machine is configured to run Docker.

Reviewer: See my note above about synchronizing these.

## Installing the Observability Pipelines Worker

{{< tabs >}}
{{% tab "Docker" %}}

The Observability Pipelines Worker Docker image is published to Docker Hub [here][1].

1. Download the [sample pipeline configuration file][2].

2. Run the following command to start the Observability Pipelines Worker with Docker:
    ```
    docker run -i -e DD_API_KEY=<API_KEY> \
      -e DD_OP_PIPELINE_ID=<PIPELINE_ID> \
      -e DD_SITE=<SITE> \
      -e SPLUNK_ENDPOINT=<SPLUNK_URL> \
      -e SPLUNK_TOKEN=<SPLUNK_TOKEN> \
      -p 8088:8088 \
      -v splunk.yaml:/etc/observability-pipelines-worker/pipeline.yaml:ro \
      datadog/observability-pipelines-worker run
    ```

>     Where `splunk.yaml` is the sample configuration you downloaded in the previous step. Be sure to update `SPLUNK_ENDPOINT` and `SPLUNK_TOKEN` with values that match your Splunk deployment you created in [Setting up the Splunk Index](#setting-up-the-splunk-index).

Reviewer 2: ```suggestion
   `splunk.yaml` is the sample configuration you downloaded in Step 1. Be sure to update `SPLUNK_ENDPOINT` and `SPLUNK_TOKEN` with values that match the Splunk deployment you created in [Setting up the Splunk Index](#setting-up-the-splunk-index).
```

    sudo systemctl restart observability-pipelines-worker
    ```

[1]: /resources/yaml/observability_pipelines/splunk/splunk.yaml
{{% /tab %}}
{{< /tabs >}}

### Load balancing

{{< tabs >}}
{{% tab "Docker" %}}

> No built-in support for load-balancing is provided, given the single-machine nature of the installation. You will need to provision your own load balancers using whatever your company's standard is.

Reviewer: See above about noting that this is not a supported production configuration.

{{< tabs >}}
{{% tab "Docker" %}}

> Production-oriented setup is not covered by these Docker instructions, and you should defer to your company's standards for load balancing in containerized environments. If you are testing on your local machine, configuring a load balancer is unnecessary.

Reviewer 2: ```suggestion
Production-oriented setup is not included in the Docker instructions. Instead, refer to your company's standards for load balancing in containerized environments. If you are testing on your local machine, configuring a load balancer is unnecessary.
```

Observability Pipelines includes multiple buffering strategies that allow you to increase the resilience of your cluster to downstream faults. The provided sample configurations use disk buffers, the capacities of which are rated for approximately 10 minutes of data at 10Mbps/core for Observability Pipelines deployments. That is often enough time for transient issues to resolve themselves, or for incident responders to decide what needs to be done with the observability data.

{{< tabs >}}
{{% tab "Docker" %}}

> By default, the Observability Pipelines Worker's data directory is set to `/var/lib/observability-pipelines-worker`. Please ensure that your host machine has a sufficient amount of storage capacity allocated to the container's mountpoint.

Reviewer 2: ```suggestion
By default, the Observability Pipelines Worker's data directory is set to `/var/lib/observability-pipelines-worker`. Make sure that your host machine has a sufficient amount of storage capacity allocated to the container's mountpoint.
```

{{% /tab %}}
{{< /tabs >}}

## Connect Splunk forwarders to the Observability Pipelines Worker
After you install and configure the Observability Pipelines Worker to send logs to your Splunk index, you must update your existing collectors to point to the Worker.

> You can update most Splunk collectors with the IP/URL of the host (or load balancer) associated with the Observability Pipelines worker.

Reviewer 2: ```suggestion
You can update most Splunk collectors with the IP/URL of the host (or load balancer) associated with the Observability Pipelines Worker.
```

> Additionally, you must update the Splunk collector with the HEC token you wish to use for authentication, so it matches the one specified in the OP Worker's list of `valid_tokens` in your `pipeline.yaml`.

Reviewer 2: ```suggestion
Additionally, you must update the Splunk collector with the HEC token you wish to use for authentication, so it matches the one specified in the Observability Pipelines Worker's list of `valid_tokens` in `pipeline.yaml`.
```

```
# Example pipeline.yaml splunk_receiver source
sources:
  splunk_receiver:
    type: splunk_hec
    address: 0.0.0.0:8088
    valid_tokens:
        - ${SPLUNK_TOKEN}
```

> In the sample configuration we have provided, we simply use the same HEC token for the Splunk source and destination.

Reviewer 2: ```suggestion
In the sample configuration provided, the same HEC token is used for both the Splunk source and destination.
```
